odoo/models.py

```diff
@@ -5533,16 +5578,18 @@ def diff(self, other):
                 if field_onchange.get(name):
                     record._onchange_eval(name, field_onchange[name], result)
 
-                # make a snapshot (this forces evaluation of computed fields)
-                snapshot1 = Snapshot(record, nametree)
+        # make a snapshot (this forces evaluation of computed fields)
+        snapshot1 = Snapshot(record, nametree)
 
-                # determine which fields have been modified
-                for name in nametree:
-                    if snapshot1[name] != snapshot0[name]:
-                        todo.append(name)
+        # determine which fields have been modified
+        for name in nametree:
+            if snapshot1[name] != snapshot0[name]:
+                todo.append(name)
```

Comment:
You have just broken the "cascading" of "classical" onchanges: if A has been changed, and its onchange method modifies B, you have to execute the onchanges of B.